### ISSUE TITLE
Role values being assigned to wrong fields

### ISSUE THREAD

**Issue by m5lapp** [splunk/terraform-provider-scp#35]
Hi there,

### Terraform Version
1.5.7

### Affected Resource(s)
Please list the resources as a list, for example:
- scp_roles

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl
resource "scp_roles" "test_ba" {
  name                          = "test_ba"
  cumulative_rt_srch_jobs_quota = 0
  cumulative_srch_jobs_quota    = 50
  default_app                   = "search"
  rt_srch_jobs_quota            = 0
  srch_jobs_quota               = 3
  srch_disk_quota               = 250
  srch_filter                   = "*"
  srch_indexes_allowed          = ["*"]
  srch_indexes_default          = ["nonprod"]
  srch_time_earliest            = -1
  srch_time_win                 = -1
  federated_search_manage_ack   = "N"
  capabilities                  = var.default_role_capabilities
}
```

### Debug Output
https://gist.github.com/m5lapp/4350d6ae3d7cedcd9bb2173dc97da32f

### Panic Output
N/A

### Expected Behavior
The role is created with the values specified above.

### Actual Behavior
After creating the role, we ran terraform apply for a second time expecting no changes to be detected. Instead, as per the linked Gist, Terraform wants to update the fields cumulative_rt_srch_jobs_quota, rt_srch_jobs_quota, srch_indexes_allowed and srch_indexes_default.

When the Terraform was first applied, it seems that:
 * The cumulative_rt_srch_jobs_quota value of 0 was not applied and we got the default of 100
 * The rt_srch_jobs_quota value of 0 was not applied and we got the default value of 6
 * The srch_indexes_allowed field has the value of the srch_indexes_default field
 * The srch_indexes_default field was not set

Looking in Splunk, we can see for the role that gets created, these four values are indeed incorrectly set.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform plan`
2. `terraform apply`
3. `terraform plan`

### Important Factoids
The issue with the srch_indexes_allowed field getting the srch_indexes_default value seems to be happening on this line:
https://github.com/splunk/terraform-provider-scp/blob/c5549ba75146241b4fb216f493015c88feef4992/internal/roles/role_resource.go#L405

We're not sure where in the code the other issues are coming from.

### References
N/A

**Comment by mausam-patel** [comment]
After triaging, the issue with `cumulative_rt_srch_jobs_quota` and `rt_srch_jobs_quota` is an issue with Terraform Plugin SDK being unable to get zero values. The SDK seems to interpret zero values as null, and fails [here](https://github.com/splunk/terraform-provider-scp/blob/c5549ba75146241b4fb216f493015c88feef4992/internal/roles/role_resource.go#L352) 

`        if value, ok := d.GetOk(schemaKeyCumulativeRTSrchJobsQuota); ok {` 

GetOk fails where there is a zero value provided for TypeInt fields: https://github.com/hashicorp/terraform-plugin-sdk/blob/main/helper/schema/resource_data.go#L87C1-L88C1 

Currently assessing if there is a workaround we may implement that is also non-breaking. In the meantime, please confirm whether a non-zero value could satisfy your configuration as a user workaround.

**Comment by m5lapp** [comment]
Hi @mausam-patel, thanks for getting back to us. Ahhh OK, I see there is a [GetExistsOK](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema#ResourceData.GetOkExists) function, but it is already deprecated unfortunately. For now we can work around the issue with the two zero values by just updating the values in the Splunk GUI.

Regarding the  srch_indexes_allowed/srch_indexes_default fields, that should be a simple change, will that be resolved soon? We had a go at fixing it ourselves but struggled to get the tests running properly locally. We also saw you do not accept contributions for this project, is that correct?

**Comment by mausam-patel** [comment]
Yes @m5lapp, fix should be out soon. First, seems like [GetChange](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema#ResourceData.GetChange) may be used for the zero value issue, doing some testing to confirm so I can release a fix with both changes at once.

**Comment by m5lapp** [comment]
Great, thanks @mausam-patel, we really appreciate your help with this.

**Comment by mausam-patel** [comment]
Fix released with v1.2.1

**Comment by m5lapp** [comment]
Hi @mausam-patel, thanks for that. I have now tested and can confirm that both the issues raised are now resolved. Thank you.
